pbsmetrics/metrics.go

```diff
@@ -272,4 +274,5 @@ type MetricsEngine interface {
 	RecordStoredReqCacheResult(cacheResult CacheResult, inc int)
 	RecordStoredImpCacheResult(cacheResult CacheResult, inc int)
 	RecordPrebidCacheRequestTime(success bool, length time.Duration)
+	RecordRequestQueueTime(requestStatus RequestStatus, requestType RequestType, length time.Duration)
```

Comment:
I think request type is good. It might be interesting to observe differences between request types which might reveal interesting insights. I don't think RequestStatus is appropriate here though. It has high cardinality and we really just care about "accepted" or "rejected" from a timing perspective. I recommend changing that to a bool.

Reply:
We now have 3 states: 
`RequestStatusErr` - in case request timeout headers are incorrect (wrong format)
`RequestStatusQueueTimeout` - in case requests stays too long in queue
and `RequestStatusOK`
Do you want me to refactor it and "merge" 2 error statuses into one?

Reply:
I don't think we need to record any metrics for the  `RequestStatusErr` state. If the headers are wrong, we won't be able to compute the duration, which leaves just timeout and ok.

The existing code will record the `RequestStatusErr` state in the request count metrics, so we know how often it happens.

I'm also fine to preload the video type only for now. Please add a comment to in the prometheus.go code since all other metrics are fully enumerated in the preload.

Reply:
Done!